packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -681,6 +683,91 @@ class ListView extends BoxScrollView {
     cacheExtent: cacheExtent
   );
 
+  /// Creates a fixed-length scrollable linear array of list "items" separated
```

Comment:
IIUC this is essentially a tweak to ListView.builder, and we can't just add a seperatorBuilder parameter to ListView.builder because then itemExtent won't make sense?

If so:
1.  should we allow infinite lists here just like we do in ListView.builder?
2.  would it make sense to keep builder in the name? maybe separatedBuilder?
3.  not for this CL but sounds like itemExtent makes things more efficient, would it make sense to eventually keep `itemExtent` and `separatorExtent`, I guess leaving a TODO for that should be fine for now.
4. actually if we plan to eventually keep separatorExtent, would it make sense to have `ListView.builder` optionally take separatorExtent and separatorBuilder?

Reply:
This is a separate constructor because itemExtent and infinite lists are not allowed. 

I considered doing exactly what you're suggesting. It's possible but surprisingly complicated: @Hixie talked me out of it.

I believe that, even with these restrictions, this constructor addresses a common case. Infinite lists aren't that common, and - for short lists - the the efficiency gained by specifying itemExtent, when it's even possible to have fixed height items, isn't huge.

Reply:
Note also: infinite lists aren't allowed because it's too hard to decide when you're at the last item. The last item is defined by the itemBuilder returning null and there's no simple way to telegraph the fact that the itemBuilder returned null to the separatorBuilder's caller.